src/main/java/com/aws/greengrass/ipc/GetClientDeviceAuthTokenOperationHandler.java

```
@@ -0,0 +1,129 @@
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final String MQTT_CREDENTIAL_TYPE = "mqtt";
+    private static final String NO_DEVICE_CREDENTIAL_ERROR = "Client device credential document is required";
+    private static final String NO_VALID_MQTT_CREDENTIAL_ERROR = "Valid mqtt credential document is required";
+    private final String serviceName;
+    private final AuthorizationHandler authorizationHandler;
+    private final SessionManager sessionManager;
+    private final Map<String, String> credentialMap = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param context              operation continuation handler
+     * @param sessionManager       session manager
+     * @param authorizationHandler authorization handler
+     */
+    public GetClientDeviceAuthTokenOperationHandler(
+            OperationContinuationHandlerContext context,
+            SessionManager sessionManager,
+            AuthorizationHandler authorizationHandler
+    ) {
+
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.sessionManager = sessionManager;
+        this.authorizationHandler = authorizationHandler;
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public GetClientDeviceAuthTokenResponse handleRequest(GetClientDeviceAuthTokenRequest request) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForClientDevAuthToken();
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            Map<String, String> credentialMap = mapOfMqttCredential(request.getCredential());
+            try {
+                String sessionId = sessionManager.createSession(MQTT_CREDENTIAL_TYPE, credentialMap);
```

catch all exceptions from createSession and throw a nice-ish error that we couldn't create a session